beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -174,20 +174,39 @@ func (s *Service) writeBlockRangeToStream(ctx context.Context, startSlot, endSlo
 			break
 		}
 	}
+
+	var reconstructedBlock []interfaces.SignedBeaconBlockWriteOnly
 	if blindedExists {
-		reconstructedBlks, err := s.cfg.executionPayloadReconstructor.ReconstructFullBellatrixBlockBatch(ctx, blks[blindedIndex:])
+		reconstructedBlock, err = s.cfg.executionPayloadReconstructor.ReconstructFullBellatrixBlockBatch(ctx, blks[blindedIndex:])
 		if err != nil {
 			log.WithError(err).Error("Could not reconstruct full bellatrix block batch from blinded bodies")
 			s.writeErrorResponseToStream(responseCodeServerError, p2ptypes.ErrGeneric.Error(), stream)
 			return err
 		}
-		copy(blks[blindedIndex:], reconstructedBlks)
 	}
 
 	for _, b := range blks {
 		if err := blocks.BeaconBlockIsNil(b); err != nil {
 			continue
 		}
+		if b.IsBlinded() {
```

Comment:
This seems like a non-cosmetic change here

Reply:
Since reconstructing blind block returns writeable bock, we have to make another loop. One loop for the read-only block and one loop for the writeable block. He didn’t have a comment after this, ill copy the same msg over